wfl/calculators/utils.py

```diff
@@ -128,7 +129,11 @@ def save_results(atoms, properties, results_prefix=None):
         except PropertyNotImplementedError:
             config_results['energy'] = atoms.get_potential_energy()
     if 'stress' in properties:
-        config_results['stress'] = atoms.get_stress()
+        # Quantum Espresso doesn't calculate stress, even if asked for, if pbc=False.
+        try:
+            config_results['stress'] = atoms.get_stress()
+        except PropertyNotImplementedError:
+            warnings.warn(f'"stress" was asked for, but not found in results.')
```

Comment:
This came about because of one of the tests - one of the structures is periodic, one is not. The old interface, and so the current one, looks at whether the stress is asked for in `properties` and removes it if it is not appropriate (e.g. the structure is non-periodic), but only at the `calculate()` step. So come this getting results stage, `parameters` include stress, but it's not in results, so error is thrown. But stress is not in the results, because it was expectedly taken out in the `calculate()` stage? That's why I have added a warning. 

I don't have experience fitting to periodic structures - would you ever have mixture of periodic/non-periodic structures, with and without stresses in your fitting/etc datasets?

Reply:
> This came about because of one of the tests - one of the structures is periodic, one is not. The old interface, and so the current one, looks at whether the stress is asked for in `properties` and removes it if it is not appropriate (e.g. the structure is non-periodic), but only at the `calculate()` step. So come this getting results stage, `parameters` include stress, but it's not in results, so error is thrown. But stress is not in the results, because it was expectedly taken out in the `calculate()` stage? That's why I have added a warning.

Hmm.  Let me think about whether the `properties` are specified in the wrong place, or if this is just inevitable. I don't really mind it, as long as it's documented that the routine can sometimes fail to calculate specific properties for good reasons, so the user knows they have to check.

> I don't have experience fitting to periodic structures - would you ever have mixture of periodic/non-periodic structures, with and without stresses in your fitting/etc datasets?

Unfortunately, yes, definitely.